_posts/2024-12-30-debezium-oracledb-logminer.md

```diff
@@ -0,0 +1,313 @@
+  2.  Download the Confluent Platform KRaft all-in-one Docker Compose file using command:
+
+      $> wget https://raw.githubusercontent.com/confluentinc/cp-all-in-one/7.8.0-post/cp-all-in-one-kraft/docker-compose.yml
+
+
+  3. Update docker-compose.yaml file with below changes:
+
+       kafka-connect:
+        image: debezium/connect:2.7.3.Final
+        hostname: kafka-connect
+        container_name: kafka-connect
+        depends_on:
+          - broker
+          - schema-registry
+        ports:
+          - "8083:8083"
+        environment:
+          BOOTSTRAP_SERVERS: 'broker:29092'
+          GROUP_ID: 1 
+          CONFIG_STORAGE_TOPIC: "my_connect_configs"
+          OFFSET_STORAGE_TOPIC: "my_connect_offset"
+          STATUS_STORAGE_TOPIC: "my_connect_status"
+          #KAFKA_CLASSPATH: /opt/oracle/ojdbc8.jar
+          LD_LIBRARY_PATH: /kafka/instantclient_21_16
+        volumes:
+            - ./basic/instantclient_21_16:/kafka/instantclient_21_16
+    
+      oracle-db:
+        image: heartu41/oracle19c
+        container_name: oracle
+        ports:
+          - "1521:1521"
+        environment:
+          ORACLE_PWD: "Oracle123"
+          ORACLE_SID: "ORCLCDB"
+          ENABLE_ARCHIVELOG: true
+          ENABLE_FORCE_LOGGING: true
+    
+        volumes:
+        - ./oracleDB/oradata:/opt/oracle/oradata
+  
+  4.  Create directory “oracleDB/oradata and change ownership to 54321
+        '''
+        $> mkdir -p oracleDB/oradata
+        $> sudo chown -R oracleDB/oradata 54321:54321
+        '''
+  5.  Start Confluent Platform stack in detach mode:
+        
+        '''
+        $> docker compose up -d
+          Each component of Confluent Platform starts in separate container.
+          
+        Creating broker   ... done
+        Creating schema-registry  ... done
+        Creating rest-proxy       ... done
+        Creating connect          ... done
+        Creating ksqldb-server    ... done
+        Creating control-center   ... done
+        Creating ksql-datagen     ... done
+        Creating ksqldb-cli       ... done
+        Creating oracle           ... done
+        
+        '''
+
+        You can verify if all services are up and running using command “docker compose ps -a”
+
+# Configuring Oracle Database
+
+  When Oracle DB container starts for the first time, there are no initial configuration and database exists. Hence database will be installed and configuration will be started. This process may take 10-20 mins (approx). You can verify readiness of Oracle DB from container logs (docker compose logs oracle) and should see message:
+
+    '''
+    #############################
+    DATABASE IS READY TO USE!
+    #############################
+    '''
+
+
+  In order to incorporate changes from an Oracle database, a number of database configurations are required:
```

Comment:
Can we have a high level description of all the steps and then go into each step? It will always good to show the readers what needs to be done to get a sense of the amount of work needed.